mtcpclient/calculate.go

```diff
@@ -38,26 +34,21 @@ func (gc GroupOfConnections) calculateMetricsReport(status tcpclient.ConnectionS
 				mr.total += item.GetTCPProcessingDuration()
 			}
 		}
+		mr.avg = mr.total / time.Duration(mr.numberOfConnections)
+		mr.stdDev = gc.calculateStdDev(mr)
 	}
-	mr.avg = mr.total / time.Duration(mr.numberOfConnections)
-	mr.stdDev = gc.calculateStdDev(status, mr)
 	return mr
 }
 
-func (gc GroupOfConnections) calculateStdDev(status tcpclient.ConnectionStatus, mr metricsCollectionStats) time.Duration {
+func (gc GroupOfConnections) calculateStdDev(mr metricsCollectionStats) time.Duration {
```

Comment:
consider the comment in the TODO. If your branch does not have tests to give you confidence, take a look to the PR in parallel ;) . If you think this change may take quite long, then discard this (it's better not to delay the merge of the rest of this PR)

Reply:
👍